samples/Kconfig

```diff
@@ -16,6 +16,14 @@ config NCS_SAMPLES_DEFAULTS
 	help
 	  Use the default configuration for NCS samples.
 
+# Attempt to default to newlibc nano, if newlibc is selected.
+# The nano flavour used to be the default newlibc flavour before NCS v2.2.0,
+# when it changed to vanilla, which has a much larger FLASH footprint
+config NCS_NRF9160_CLIB_SIZE_OPTIMIZATION
+	bool
+	default y if SOC_SERIES_NRF91X
+	imply NEWLIB_LIBC_NANO if NEWLIB_LIBC
+
```

Comment:
no reason for a new kconfig, and certainly not with `imply`.

Let's just change the default to be nano in Kconfig.nrf here:
https://github.com/nrfconnect/sdk-nrf/blob/cce7127811522aae923f2ef062053601a5e0ed79/Kconfig.nrf#L10

Add something like:
```
# Zephyr default is newlib and not newlib-nano for consistency between architectures.
# For footprint reasons, newlib-nano is preferred default in NCS when newlib is selected.
config NEWLIB_LIBC_NANO
	default y
	depends on NEWLIB_LIBC && HAS_NEWLIB_LIBC_NANO
```